till_korten/devbio-napari_cluster/readme.md

```diff
@@ -0,0 +1,214 @@
+* you have a working python script or jupyter notebook that processes your data
+* the computers available to you limit you because:
+  * lacking GPU (or too little GPU RAM)
+  * not enough RAM
+  * not enough disk space
+  * not enough CPUs
+
+### You may need to look elsewhere if:
+* you are still actively developing on your workflow and are installing/removing python packages on a regular basis. We are working with singularity images and it is not feasible to frequently build new images for you with new python packages. You may want to look at Roberts article on [using google colab](../../robert_haase/clesperanto_google_colab/).
+* your workflow needs a graphical user interface other than standard jupyter notebook stuff
+
+### See also
+* [ZIH HPC Documentation](https://doc.zih.tu-dresden.de/)
+
+## Step 1: Get access to the ZIH cluster
+Fill out a project proposal and submit it to the compute center as [explained in their documentation](https://tu-dresden.de/zih/hochleistungsrechnen/zugang/projektantrag?set_language=en).
+
+## Step 2: Start a Jupyter session on the ZIH cluster
+You will be greeted with the TUD login screen. Log in with your ZIH user name and password:
+
+<img src="images/1_login.png" width="500" />
+
+
+Afterwards, you should see a single button `Start My Server`. Click on it:
+
+<img src="images/2_start_server.png" width="500" />
+
+Now you get to configure the computing node you want your session to run on. Switch to the Advanced configuration by clicking the button `Advanced`. Then you should see something like the image below. 
+
+Start by choosing a preset (click on 1). You should choose a GPU node preset (2). You can choose either a n Ampere A100 or a Tesla K80 card. Choose according to the your needs and the current utilization (the bars above the preset chooser) (the A100 are on `alpha`, the K80 are on `gpu2`) If the partition is very full, you may have to wait a long time or not get a session at all. Choose the number of CPUs and memory per cpu as needed. Note that the memory is per CPU, so if you choose more CPUs, you get more memory.
+
+Once you are happy with your configuration, click the orange button `Spawn` at the very bottom.
+
+<img src="images/3_configure_node.png" width="500" />
+
+You will now see a wait bar. Do not worry if it does not move, this bar is always at 50%. It usually takes 2-5 min to get a node.
+
+<img src="images/4_wait.png" width="500" />
+
+## Step 3: Open a terminal
+
+Open a terminal by clicking on file (1 in the image below) -> new (2) -> Terminal (3)
+
+<img src="images/5_open_terminal.png" width="500" />
+
+## Step 4: Install a custom jupyter kernel for your user
+
+In the terminal, execute the following code:
+
+```bash
+git clone https://gitlab.mn.tu-dresden.de/bia-pol/singularity-devbio-napari.git
+```
+You will be asked for a user name and password. Enter the following:
+
+User name: `git_pull`
+
+Password: `Bc6Yg8k5MyZzURNuWChR`
```

Comment:
This is a token that has only pull access to the repository. Without it, people could not get the files they need to install the singularity containers and jupyter kernels.

From a security point of view, it is like making just the git the repository public. Without giving users the ability to all the other gitlab functionality (like reading/posting issues etc).

In the mid-term it would make more sense, to actually make the project public. Then we can simply revoke the token and remove it here.